pypsa-to-iamc

```diff
@@ -0,0 +1,634 @@
+              var[sh][sheet_var[sheet]+'|Electricty|Biomass|CCS'] = var[sh][sheet_var[sheet]+'|Electricity|Solid bio and waste|CCS']            
+           
+              #Energy generation : Hydrogen
+            
+              var[sh][sheet_var[sheet]+'|Electricity|Gases|Hydrogen'] = 0.001*ct*gen_des(time((n.links_t.p1.filter(like ='H2 Fuel Cell').filter(like =country)).sum(axis=1)))
+
+              #Energy generation : Oil 
+
+              var[sh][sheet_var[sheet]+'|Electricity|Liquids|Fossil'] = 0.001*ct*gen_des(time((n.links_t.p1.filter(like ='oil-').filter(like =country)).sum(axis=1)))
+
+              #Energy generation : Power to gas 
+              
+              var[sh][sheet_var[sheet]+'|P2G|Electricity'] = 0.001*ct*gen_des(time(n.links_t.p1.filter(like ='Sabatier').filter(like =country).sum(axis=1)))
+              var[sh][sheet_var[sheet]+'|P2G|Electricity']+= 0.001*ct*gen_des(time(n.links_t.p1.filter(like ='helmeth').filter(like =country).sum(axis=1))) 
+              var[sh][sheet_var[sheet]+'|P2G|Electricity']+= 0.001*ct*gen_des(time(n.links_t.p1.filter(like ='H2 Electrolysis').filter(like =country).sum(axis=1)))
+              var[sh][sheet_var[sheet]+'|P2G|Electricity']+= 0.001*ct*gen_des(time(n.links_t.p1.filter(like ='Fischer-Tropsch').filter(like =country).sum(axis=1)))               
+              
+            
+              #Energy generation : hydro (reservoir, ror)
+
+              var[sh][sheet_var[sheet]+'|Electricity|Hydro|river'] = 0.001*ct*gen_des(time(n.generators_t.p.filter(like ='ror').filter(like =country).sum(axis=1)))
+              var[sh][sheet_var[sheet]+'|Electricity|Hydro|dam'] = 0.001*ct*gen_des(time(n.storage_units_t.p.filter(like ='hydro').filter(like =country).sum(axis=1)))
+              var[sh][sheet_var[sheet]+'|Electricity|Hydro'] = var[sh][sheet_var[sheet]+'|Electricity|Hydro|river']+var[sh][sheet_var[sheet]+'|Electricity|Hydro|dam']
+           
+              #Energy generation : storage (PHS, battery, H2 storage)
+
+              var[sh][sheet_var[sheet]+'|Flexibility|Electricity Storage|Medium duration'] = 0.001*ct*(gen_des(time(n.storage_units_t.p.filter(like ='PHS').filter(like =country).sum(axis=1)))+\
+               gen_des(time(n.stores_t.p.filter(like ='H2').filter(like =country).sum(axis=1))))
+            
+              var[sh][sheet_var[sheet]+'|Flexibility|Electricity Storage|Short duration'] = 0.001*ct*(gen_des(time(n.stores_t.p.filter(like ='battery').filter(like =country).sum(axis=1)))+\
+               gen_des(time(n.stores_t.p.filter(like ='Li ion').filter(like =country).sum(axis=1))))
+            
+              var[sh][sheet_var[sheet]+'|Flexibility|Electricity Storage'] =  (var[sh][sheet_var[sheet]+'|Flexibility|Electricity Storage|Medium duration']
+               +var[sh][sheet_var[sheet]+'|Flexibility|Electricity Storage|Short duration'])
```

Comment:
We need to be careful here. Storage technologies do not have net energy generation!
If you sum stores_t.p , don't you get zero?
If you want to calculate the load coverage, you should sum only the positive part or the power of the link discharging the storage.

Reply:
Also, should I do this for storage_units as well?

Reply:
Added the positive-only values for both stores and storages units.